packages/mobile/src/components/details-tile/DetailsTileHasAccess.tsx

```diff
@@ -231,61 +243,38 @@ export const DetailsTileHasAccess = ({
       )
     }
     if (followee) {
-      return (
-        <View style={styles.descriptionContainer}>
-          <Text>
-            <Text style={styles.description}>
-              {messages.unlockedFollowGatedPrefix}
-            </Text>
-            <Text style={styles.description}>{followee.name}</Text>
-            <UserBadges
-              badgeSize={16}
-              user={followee}
-              nameStyle={styles.description}
-              hideName
-            />
-            <Text style={styles.description}>
-              {messages.unlockedFollowGatedSuffix}
-            </Text>
-          </Text>
-        </View>
-      )
+      return renderUnlockedSpecialAccessDescription({
+        entity: followee,
+        prefix: messages.unlockedFollowGatedPrefix,
+        suffix: messages.unlockedFollowGatedSuffix
+      })
     }
     if (tippedUser) {
-      return (
-        <View style={styles.descriptionContainer}>
-          <Text>
-            <Text style={styles.description}>
-              {messages.unlockedTipGatedPrefix}
-            </Text>
-            <Text style={styles.description}>{tippedUser.name}</Text>
-            <UserBadges
-              badgeSize={16}
-              user={tippedUser}
-              nameStyle={styles.description}
-              hideName
-            />
-            <Text style={styles.description}>
-              {messages.unlockedTipGatedSuffix}
-            </Text>
-          </Text>
-        </View>
-      )
+      return renderUnlockedSpecialAccessDescription({
+        entity: tippedUser,
+        prefix: messages.unlockedTipGatedPrefix,
+        suffix: messages.unlockedTipGatedSuffix
+      })
     }
 
     console.warn(
       'No entity for premium conditions... should not have reached here.'
     )
     return null
-  }, [nftCollection, followee, tippedUser, handlePressCollection, styles])
+  }, [
```

Comment:
Not a pattern you introduced, but it's a little difficult to parse out what this dependency array is tied to. We might want to split the content of these render functions into separate components so that you can see the `renderX() = useCallback( () => { ...}, [dependencies])` a little more easily.